src/jrd/tra.cpp

```diff
@@ -1948,7 +1948,7 @@ void TRA_sweep(thread_db* tdbb)
 }
 
 
-int TRA_wait(thread_db* tdbb, jrd_tra* trans, TraNumber number, jrd_tra::wait_t wait)
+int TRA_wait(thread_db* tdbb, jrd_tra* trans, TraNumber number, jrd_tra::wait_t wait, Nullable<SSHORT> waitTimeout)
```

Comment:
This looks like redundant change: the only case when `waitTimeout` is passed here is `VIO\wait()` and the passed value is 0. It might be better to change `VIO\wait()` to pass `tra_no_wait` instead and not change `TRA_wait()` at all. 
As for me, there must be more strong reason to change `TRA` interface.

Reply:
What seems to work is `tra_probe`.

Reply:
Exactly!